v2/pkg/protocols/others/ssl/ssl.go

```diff
@@ -0,0 +1,180 @@
+
+	if r.options.Options.Debug || r.options.Options.DebugRequests {
+		gologger.Info().Str("address", input).Msgf("[%s] Dumped SSL request for %s", r.options.TemplateID, input)
+	}
+
+	state := connTLS.ConnectionState()
+	if len(state.PeerCertificates) == 0 {
+		return nil
+	}
+
+	tlsData := cryptoutil.TLSGrab(&state)
+	jsonData, _ := jsoniter.Marshal(tlsData)
+	jsonDataString := string(jsonData)
+
+	data := make(map[string]interface{})
+	cert := connTLS.ConnectionState().PeerCertificates[0]
+
+	data["response"] = jsonDataString
+	data["host"] = input
+	data["not_after"] = float64(cert.NotAfter.Unix())
+	data["ip"] = r.dialer.GetDialedIP(hostname)
+
+	event := eventcreator.CreateEvent(r, data, r.options.Options.Debug || r.options.Options.DebugResponse)
+	if r.options.Options.Debug || r.options.Options.DebugResponse {
+		responseOutput := jsonDataString
+		gologger.Debug().Msgf("[%s] Dumped SSL response for %s", r.options.TemplateID, input)
+		gologger.Print().Msgf("%s", responsehighlighter.Highlight(event.OperatorsResult, responseOutput, r.options.Options.NoColor))
+	}
+	callback(event)
+	return nil
+}
+
+// getAddress returns the address of the host to make request to
+func getAddress(toTest string) (string, error) {
+	if strings.Contains(toTest, "://") {
+		parsed, err := url.Parse(toTest)
+		if err != nil {
+			return "", err
+		}
+		_, port, _ := net.SplitHostPort(parsed.Host)
+
+		if parsed.Scheme == "https" && port == "" {
+			toTest = net.JoinHostPort(parsed.Host, "443")
+		} else {
+			toTest = parsed.Host
+		}
+		return toTest, nil
+	}
+	return toTest, nil
+}
+
+// Match performs matching operation for a matcher on model and returns:
+// true and a list of matched snippets if the matcher type is supports it
+// otherwise false and an empty string slice
+func (r *Request) Match(data map[string]interface{}, matcher *matchers.Matcher) (bool, []string) {
+	return protocols.MakeDefaultMatchFunc(data, matcher)
+}
+
+// Extract performs extracting operation for an extractor on model and returns true or false.
+func (r *Request) Extract(data map[string]interface{}, matcher *extractors.Extractor) map[string]struct{} {
+	return protocols.MakeDefaultExtractFunc(data, matcher)
+}
+
+// MakeResultEvent creates a result event from internal wrapped event
+func (r *Request) MakeResultEvent(wrapped *output.InternalWrappedEvent) []*output.ResultEvent {
+	return protocols.MakeDefaultResultEvent(r, wrapped)
+}
+
+// GetCompiledOperators returns a list of the compiled operators
+func (r *Request) GetCompiledOperators() []*operators.Operators {
+	return []*operators.Operators{r.CompiledOperators}
+}
+
+func (r *Request) MakeResultEventItem(wrapped *output.InternalWrappedEvent) *output.ResultEvent {
+	data := &output.ResultEvent{
+		TemplateID:       types.ToString(r.options.TemplateID),
+		TemplatePath:     types.ToString(r.options.TemplatePath),
+		Info:             r.options.TemplateInfo,
+		Type:             "ssl",
```

Comment:
The type should be an enum value and renamed to "TemplateType" or "TemplateProtocol" or smth